src/main/java/org/broadinstitute/hellbender/tools/exome/ExomeReadCounts.java

```diff
@@ -0,0 +1,881 @@
+     * <p>
+     * Keeps independent counts for each sample.
+     * </p>
+     */
+    private static final class SampleCountColumns extends CountColumns {
+
+        /**
+         * Creates a new count-counts instance given the reference to the embedding tool instance.
+         * @throws IllegalArgumentException if {@code tool} is {@code null}.
+         * @throws IllegalStateException if the current state of {@code tool} is not compatible with
+         *    this count-column manager type. See {@link #checkToolState(ExomeReadCounts)}.
+         */
+        private SampleCountColumns(final ExomeReadCounts tool) {
+            super(tool);
+        }
+
+        @Override
+        protected void checkToolState(final ExomeReadCounts tool) {
+            super.checkToolState(tool);
+            if (tool.groupBy != GroupBy.SAMPLE) {
+                throw new IllegalStateException("incompatible grouping by option value: " + tool.groupBy);
+            }
+            if (tool.sampleDB == null) {
+                throw new IllegalStateException("the sample-db should have been initialized");
+            }
+        }
+
+        @Override
+        protected String columnName(final int index) {
+            final Sample sample = tool.sampleDB.samples().get(index);
+            final String sampleId = sample.getId();
+            String columnName = CountColumnNamePattern.DEFAULT.replaceAll(tool.countColumnName,sampleId);
+            columnName = CountColumnNamePattern.COHORT.replaceAll(columnName,tool.cohortName);
+            columnName = CountColumnNamePattern.SAMPLE.replaceAll(columnName,sampleId);
+            columnName = CountColumnNamePattern.READ_GROUP.replaceAll(columnName,() -> {
+                if (sample.readGroupCount() > 1) {
+                    throw new UserException.BadArgumentValue("countColumnName", tool.countColumnName,
+                            String.format("cannot use '%s' (read-group id) in column count header when grouping by '%s' if" +
+                                    " sample with more than one read group (e.g. sample '%s' has %d).",
+                                    CountColumnNamePattern.READ_GROUP.string,tool.groupBy.friendlyName(),sample.getId(),sample.readGroupCount()));
+                }
+                return sample.readGroupCount() == 0 ? ANONYMOUS_READ_GROUP_ID : sample.readGroups().get(0);
+            });
+            return columnName;
+        }
+
+        @Override
+        protected int columnIndex(SAMRecord read) {
+            return tool.sampleDB.sampleIndexByRead(read);
+        }
+
+        @Override
+        public int columnCount() {
+            return tool.sampleDB.sampleCount();
+        }
+    }
+
+    /**
+     * Count-column manager when grouping by read-groups.
+     *
+     * <p>
+     *     Keeps independent counts per read-group.
+     * </p>
+     */
+    private static final class ReadGroupCountColumns extends CountColumns {
+
+        /**
+         * {@inheritDoc}
```

Comment:
why does everyone else (2 other constructors in sibling classes) get a javadoc and this guy only gets a {@inheritDoc}?